File: ATSScanner/Models/UserProfile.cs
using System.ComponentModel.DataAnnotations;

namespace ATSScanner.Models
{
    public class UserProfile
    {
        public int Id { get; set; }
        
        [Required]
        public int UserId { get; set; }
        
        [StringLength(50)]
        public string? FirstName { get; set; }
        
        [StringLength(50)]
        public string? LastName { get; set; }
        
        [StringLength(20)]
        public string? PhoneNumber { get; set; }
        
        [StringLength(500)]
        public string? Bio { get; set; }
        
        public DateTime? DateOfBirth { get; set; }
        
        public DateTime LastUpdated { get; set; } = DateTime.UtcNow;
        
        // Navigation property
        public User User { get; set; } = null!;
    }
}

File: client/src/types/userProfile.ts
export interface UserProfile {
  userId: number;
  username: string;
  email: string;
  firstName?: string;
  lastName?: string;
  phoneNumber?: string;
  bio?: string;
  dateOfBirth?: string;
  lastUpdated: string;
  subscription?: UserSubscription;
  scanCount: number;
  lastScanDate?: string;
}

export interface UserSubscription {
  planName: string;
  type: string;
  price: number;
  scanLimit: number;
  scansUsed: number;
  scansRemaining: number;
  isActive: boolean;
  startDate?: string;
  endDate?: string;
  stripeSubscriptionId?: string;
  canCancel: boolean;
}

export interface UpdateUserProfile {
  firstName?: string;
  lastName?: string;
  phoneNumber?: string;
  bio?: string;
  dateOfBirth?: string;
}